zkvm/src/contract.rs

```diff
@@ -52,27 +53,17 @@ enum FrozenItem {
 /// Note: values do not necessarily have open commitments. Some can be reblinded,
 /// others can be passed-through to an output without going through `cloak` and the constraint system.
 #[derive(Clone, Debug)]
-struct FrozenValue {
+pub struct FrozenValue {
     qty: Commitment,
     flv: Commitment,
 }
 
 impl Input {
-    /// Creates a "frozen contract" from payload (which is a vector of (qty, flv)) and pred.
-    /// Serializes the contract, and uses the serialized contract and txid to generate a utxo.
+    /// Serializes the contract, and uses the serialized contract and txid
```

Comment:
This is an implementation detail. Maybe just say "/// Creates an `Input` with a given `Output` and transaction id"?